### ISSUE TITLE
EIDM does not stop at position

### ISSUE THREAD

**Issue by namdre** [eclipse/sumo#8910]

**Comment by Domsall** [comment]
I took a look into the issue and found the reason again.
The problem stems from estimated variables and the way stopSpeed is calculated. Because of reaction times and estimated parameters, the IDM normally reacts too late and therefore drives a bit too far. For followSpeed I subtract 0.25m to account for this.

StopSpeed is a bit trickier. In the original IDM implementation mingap does not get added to the actual gap and the desired gap. Therefore, the gap goes to 0. With estimated variables the values do not perfectly go to 0, but "jump around 0" and dividing by small values (desired gap / gap) creates oscillating accelerations.

So even in stopSpeed I add the mingap to the actual gap and the desired gap to get smoother accelerations. The only problem is that for example at traffic lights the actual gap goes to 0 and does not become negative. When the estimated gap is a bit too high (e.g. 1.5) and the desired gap a bit lower (e.g. 1.4), then the vehicle never fully stops and crawls over the traffic light. That is why I subtract 0.75m to ensure that the vehicles always stops, but unfortunately at larger distances then.

Do you maybe have an idea how this could be fixed?

**Comment by namdre** [comment]
Arguably, estimation error is reduced at low speeds and low target distances. A small overshoot (< 0.1m) isn't problematic either. Thus drivers should always be able to stop with the needed accuracy when approaching along a smooth deceleration curve.

Other possible solutions:
- making the stop-reached criterion configurable or cfmodel dependent (could have side effects of person-can-enter-vehicle checks)
- introducing separate functions for stopSpeedAtStop and stopSpeedSomewhereElse (I'd really like to avoid this)
- warning the user if they define lane stops without sufficient slack for EIDM

**Comment by Domsall** [comment]
The estimation errors are actually pretty small when gaps or speeds reach small values, because the sigma-values are multiplied with the observed gaps and speeds. Due to the Wiener Process, the stochastic values can still be "randomly" high at some times.
I already have some safety features to neglect the estimation errors in some cases, but I did not want to suddenly neglect them at low speeds, because I then get an acceleration jump shortly before reaching the stop.

I already changed the Code a bit and will do some more tests before pushing it back
